Content.Shared/Mech/Equipment/Systems/MechSoundboardSystem.cs

```diff
@@ -0,0 +1,74 @@
+using Content.Shared.Mech;
+using Content.Shared.Mech.Equipment.Components;
+using Content.Shared.Mech.Equipment.Systems;
+using Content.Shared.Timing;
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using System.Linq;
+
+namespace Content.Shared.Mech.Equipment.Systems;
+
+/// <summary>
+/// Handles everything for mech soundboard.
+/// </summary>
+public sealed class SharedMechSoundboardSystem : EntitySystem
+{
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly UseDelaySystem _useDelay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<MechSoundboardComponent, ComponentGetState>(OnGetState);
+        SubscribeLocalEvent<MechSoundboardComponent, ComponentHandleState>(OnHandleState);
+
+        SubscribeLocalEvent<MechSoundboardComponent, MechEquipmentUiStateReadyEvent>(OnUiStateReady);
+        SubscribeLocalEvent<MechSoundboardComponent, MechEquipmentUiMessageRelayEvent>(OnSoundboardMessage);
+    }
+
+    private void OnGetState(EntityUid uid, MechSoundboardComponent comp, ref ComponentGetState args)
+    {
+        args.State = new MechSoundboardComponentState(comp.Sounds);
+    }
+
+    private void OnHandleState(EntityUid uid, MechSoundboardComponent comp, ref ComponentHandleState args)
+    {
+        if (args.Current is not MechSoundboardComponentState state)
+            return;
+
+        comp.Sounds = state.Sounds;
```

Comment:
So the footgun with state handling is if you have a ref type you need to not point at the state ref as it can cause prediction issues (e.g. you modify the field but suddenly it's modifying the compstate's one when it gets re-run). AutoNetworked stuff does it automatically but the alternatively is you just do something like .Clear() + .AddRange or the likes.

Reply:
yes i have no reason to handle state